docs/data/routes/docs/techniques/redirection/headless-ssr/en.md

```diff
@@ -0,0 +1,134 @@
+   SITECORE_API_KEY="{5379B475-A885-4B62-9A51-EE79E5434650}"
+   SITECORE_GRAPHQL_ENDPOINT="https://cm.jss16.localhost/api/jssreactweb?sc_apikey={5379B475-A885-4B62-9A51-EE79E5434650}"
+   DEFAULT_LANGUAGE="en"
+   ```
+
+2. Install the `dotenv` npm package to load the environment variables into the Node `process.env`:
+
+   ```
+   npm install --save-dev dotenv
+   ```
+
+3. In `/config.js`:
+
+   1. Require and configure `dotenv`:
+
+   ```javascript
+   require('dotenv').config();
+   ```
+
+   2. Declare the following constants: 
+
+      ```javascript
+      const graphQLEndpoint = process.env.SITECORE_GRAPHQL_ENDPOINT || serverBundle.graphQLEndpoint || `${apiHost}/api/${appName}`;
+      
+      const defaultLanguage = process.env.DEFAULT_LANGUAGE || serverBundle.defaultLanguage || 'en';
+      
+      const hostname = process.env.SITECORE_HOSTNAME || serverBundle.hostname || apiHost;
+      ```
+
+   3. For the `config` module, define a new member function `getRedirects`. This example uses a GraphQL query: 
+
+      ```javascript
+      getRedirects: () => {
+          return fetch(graphQLEndpoint, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({
+              query: `
+              query {
+                search(fieldsEqual: [{ name: "_templatename", value: "Redirect Route" }]) {
+                  results {
+                    items {
+                      item {
+                        url
+                        field(name: "Destination") {
+                          ... on LinkField {
+                            url
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            `
+            })
+          }).then(response => response.json())
+            .then(response => {
+              // Layout Service returns paths with language included. But end-user may not have
+              // language in the URL if site is in the default language. We can account for this using regular expression matching.
+              const re = new RegExp(`^(?:${hostname})(?:\/${defaultLanguage})?(\/.+)$`);
+              function getNormalizedPath(url) {
+                return re.test(url) ? url.match(re)[1] : url;
+              }
+      
+              return response.data.search.results.items.reduce((redirects, searchResult) => {
+                const source = searchResult.item?.url;
+                const destination = searchResult.item?.field?.url;
+      
+                if (source && destination) {
+                  redirects[getNormalizedPath(source)] = getNormalizedPath(destination);
+                }
+      
+                console.log(JSON.stringify(redirects, null, 4));
+                return redirects;
+              }, {});
+            });
+        }
+      ```
+
+4. In `/index.js`, leverage the new `getRedirects` method: 
```

Comment:
Link to index.js: https://github.com/Sitecore/jss/blob/master/samples/node-headless-ssr-proxy/index.js

Reply:
Same as above. They should be working in their own directory and therefore have a local copy of the file.